src/jitlayers.cpp

```diff
@@ -644,6 +649,24 @@ class JLDebuginfoPlugin : public ObjectLinkingLayer::Plugin {
 }
 
 # ifdef LLVM_SHLIB
+#  if JL_LLVM_VERSION >= 140000
+class JLEHFrameRegistrar final : public jitlink::EHFrameRegistrar {
+public:
+    Error registerEHFrames(orc::ExecutorAddrRange EHFrameSection) override {
+        register_eh_frames(
+            reinterpret_cast<uint8_t *>(EHFrameSection.Start.getValue()),
```

Comment:
I would recommend using `EHFrameSection.Start.toPtr<uint8_t *>()` here and in deregistering for terseness/clarity